### PR TITLE
Web3inbox fix 500

### PR DESCRIPTION
**Pull request by thegodfatherbeeearth**
1. Don't return 500 error in all calls to the frontend's /api/notify
2. Send web3inbox notifications on x-chain & multilink claiming

### CODE REVIEW

File: src/app/api/notify/route.ts [hunk 1]
```diff
@@ -28,6 +28,11 @@ export async function POST(request: NextRequest, response: NextResponse) {
             body: JSON.stringify(notificationPayload),
         })
     } catch (error: any) {
+        console.error('Error occured while sending notification:', error)
         throw new Error('Error occured while sending notification')
     }
+
+    return NextResponse.json({
+        status: 200,
+    })
```
Comment (Hugo0) [inline src/app/api/notify/route.ts]: Why is this necessary? Because fetch() needs a 200 response else it raises a 500 err? (just trying to understand)

File: src/utils/notifications.utils.ts [hunk 2]
```diff
@@ -5,8 +5,26 @@ interface INotification {
...
+export const sendNotification = async (
+    senderAddress: string,
+    recipientAddress: string | undefined,
+    chainName: string
+) => {
+    console.log('sendNotification', senderAddress)
+    const accounts = [`eip155:1:${senderAddress}` ?? '']
+    const notification: INotification = {
+        title: 'Peanut Protocol',
+        body: `Your link has been claimed on ${chainName} by ${recipientAddress ?? ''}`,
```
Comment (Hugo0) [inline src/utils/notifications.utils.ts:17]: This is not part of this PR, but imo I'd also add the USD amount / tokenamount to the message.

"your 10 usdc link ...."

Imagine if you sent multiple links, how does this tell you which one it is?

File: src/components/claim/views/success.view.tsx [hunk 3]
```diff
@@ -35,25 +35,12 @@ export function ClaimSuccessView({ txHash, claimDetails, senderAddress }: _const
...
-            body: `Your link has been claimed on ${chainName} by ${address ?? ''}`,
-            url: undefined,
-            type: '2aee6e5f-091d-444e-96cd-868ba2ddd0e7',
-        }
-        utils.sendNotification({
-            notification,
-            accounts,
-        })
+    const sendNotification = async () => {
+        const chainName = chainDetails.find((detail) => detail.chainId === claimDetails[0].chainId)?.name
```
Comment (Hugo0) [inline src/components/claim/views/success.view.tsx:42]: Given that chainDetails doesn't update on success view, and this line is repeated: `chainDetails.find((detail) => detail.chainId === claimDetails[0].chainId)?.name`

Why not make it a const?

File: src/components/claim/multilinkViews/multilinkSuccess.view.tsx [hunk 4]
```diff
@@ -23,8 +25,14 @@ export function multilinkSuccessView({ txHash, claimDetails }: _consts.IClaimScr
 
     useEffect(() => {
         router.prefetch('/send')
+        sendNotification()
     }, [])
 
+    const sendNotification = async () => {
+        const chainName = chainDetails.find((detail) => detail.chainId === claimDetails[0].chainId)?.name
```
Comment (Hugo0) [inline src/components/claim/multilinkViews/multilinkSuccess.view.tsx:32]: sweet! thanks for thinking of multilinks in regards to web3inbox! good thinking

File: src/utils/notifications.utils.ts (hunk 2 above)
Comment (Hugo0) [reply src/utils/notifications.utils.ts:17]: Also, good on splitting this out, since we have multiple success views. Sweet.

File: src/app/api/notify/route.ts (hunk 1 above)
Comment (thegodfatherbeeearth) [reply src/app/api/notify/route.ts]: I think it's how nextjs works, not related to `fetch`. Function must return something successful, otherwise next thinks something went wrong and returns a 500 to the caller

File: src/utils/notifications.utils.ts (hunk 2 above)
Comment (thegodfatherbeeearth) [reply src/utils/notifications.utils.ts:17]: Yeah, agreed! I was also thinking about this. Created a backlog task with owner = Bo

File: src/components/claim/views/success.view.tsx (hunk 3 above)
Comment (thegodfatherbeeearth) [reply src/components/claim/views/success.view.tsx:42]: Honestly, dunno. From what I've found, chainDetails in UI is always used this way: `const [chainDetails] = useAtom(store.defaultChainDetailsAtom)`, and I thought it's better to not make any changes to the approach without Bo.

File: src/app/api/notify/route.ts (hunk 1 above)
Comment (borcherd) [reply src/app/api/notify/route.ts]: Nextjs wants all api calls to happen from the api folder. This is handles as an actual 'next api', so also expects an next api response. Thanks for introducing this alexey
